pkg/controller/replication_controller.go

```diff
@@ -257,14 +257,18 @@ func (rm *ReplicationManager) deletePod(obj interface{}) {
 		}
 	}
 	if rc := rm.getPodControllers(pod); rc != nil {
-		rm.expectations.DeletionObserved(rc)
+		rcKey, err := controllerKeyFunc(rc)
+		if err != nil {
+			glog.Errorf("Couldn't get key for replication controller %+v: %v", rc, err)
```

Comment:
If you decide to use the key function, you would modify it to take
runtime.Object types.

On Thu, Jul 9, 2015 at 9:45 AM, Ananya Kumar <notifications@github.com>
wrote:

> In pkg/controller/replication_controller.go
> <https://github.com/vmarmol/kubernetes/pull/2#discussion_r34275783>:
>
> > @@ -257,14 +257,18 @@ func (rm *ReplicationManager) deletePod(obj interface{}) {
> >  		}
> >  	}
> >  	if rc := rm.getPodControllers(pod); rc != nil {
> > -		rm.expectations.DeletionObserved(rc)
> > +		rcKey, err := controllerKeyFunc(rc)
> > +		if err != nil {
> > +			glog.Errorf("Couldn't get key for replication controller %+v: %v", rc, err)
>
> Added continue, and fixed similar issues in the replication controller.
>
> So I wanted to use the same Expectations class for both rc and dc, so I
> generalized Expectations to accept a string argument instead of an rc. I
> tried to use ObjectMeta instead, but it didn't work out, because
> KeyFunc/rcKeyFunc doesn't work on ObjectMeta, but only on API types.
>
> I guess another option is to refactor the key function, and then modify
> the Expectations class to accept ObjectMetas instead of string keys. Is
> that a better option?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/vmarmol/kubernetes/pull/2/files#r34275783>.
>